clients/web/src/lib/components/atoms/Breadcrumb/addCrumb.ts

```diff
@@ -0,0 +1,4 @@
+import { getContext, setContext } from "svelte";
+import { type ICrumb, type BreadcrumbContext, BREADCRUMB_CONTEXT_KEY } from "./types";
+
+export const addCrumb = (crumb: ICrumb) => setContext<BreadcrumbContext>(BREADCRUMB_CONTEXT_KEY, [...getContext<BreadcrumbContext>(BREADCRUMB_CONTEXT_KEY), crumb])
```

Comment:
I noticed inside the `Breadcrumb` component, we're using `hasContext` before `getContext`. Is that necessary here? Or is it not necessary in the other place?

Reply:
Good catch, it should be used here